### ISSUE TITLE
A lint to detect `///` comments unattached to anything

### ISSUE THREAD

**Issue by ghost** [rust-lang/rust-clippy#7836]
### What it does

It checks for `///` documentation comments not attached to any function, type etc. I.e. it gives a warning when the comment wouldn't cause anything to be documented.
It tells the user to remove one slash to make it a normal comment or to attach it to something and maybe say something like "make sure there's no extra newline in between!".
Note that this lint should also catch code like this:
```rs
    /// my variable called x
    let x = 0;

    /// my if checking my x 
    if x == 1 {
        // …
    }
```
Even though the documentation comments *are* attached to something, they do not cause anything to be documented (in the generated documentation I mean). In this specific case clippy could realize that it *is* attached to something and so only suggest to remove one `/`. In this case it shouldn't suggest to attach it to something because it already is.

### Categories (optional)

- Kind: `clippy::correctness` or `clippy::suspicious`

The advantage is that no documentation goes wasted/unused and things that were supposed to have documentation don't accidentally end up not being documented (and possibly confusing the developer or documenter).
It also enforces a good comment style.

### Drawbacks

None.

### Example

```rust
/// Gets you a `2`.

fn get_two() -> u32 {
    2
}
```

Could (or rather *should*) be written as:

```rust
/// Gets you a `2`.
fn get_two() -> u32 {
    2
}
```

I've also seen mistakes like these in the past where there's an extra newline instead of `///` to actually connect the two documentation comment paragraphs.
```rs
/// Gets the current year.
/// When you use the function you should note that when
/// you invoke it you don't pass any arguments
/// please because this function doesn't take any.

/// Also here's an example:
/// ```
/// let year = get_year();
/// ```
fn get_year() { ... }
```
It prevents this as well.

**Comment by r00ster91** [comment]
Just noticed there is a check like that in rustc but weirdely enough it does not fire for comments on the top level:
```rs
/// a

fn main() {
    /// b
}
```
```
error[E0585]: found a documentation comment that doesn't document anything
 --> src/main.rs:4:5
  |
4 |     /// b
  |     ^^^^^
  |
  = help: doc comments must come before what they document, maybe a comment was intended with `//`?

For more information about this error, try `rustc --explain E0585`.
```
It doesn't complain about the first comment at all.
